app/src/main/java/app/accrescent/client/workers/AutoUpdateWorker.kt

```diff
@@ -34,4 +40,24 @@ class AutoUpdateWorker @AssistedInject constructor(
 
         return Result.success()
     }
+
+    companion object {
+        private const val UPDATER_WORK_NAME = "UPDATE_APPS"
+
+        fun enqueue(context: Context, networkType: NetworkType, forceUpdate: Boolean = true) {
```

Comment:
Is there any reason we should have this `forceUpdate` parameter instead of simply always using `ExistingPeriodicWorkPolicy.UPDATE`? As is, `UPDATE` will always be used anyway.

Reply:
Oh damn, that's my bad.
In the `MainActivity` `forceUpdate` should be false so that the update checker timeout doesn't get reset when the user opens the app (so my plans).
Though I'm not really sure whether it's the intended behavior that the updater resets (so the four hours or something start again from the beginning) or not, so that depends on your preference :)

Reply:
The timer doesn't get reset with `UPDATE`. From https://developer.android.com/reference/androidx/work/ExistingPeriodicWorkPolicy#UPDATE:

> It preserves enqueue time, e.g. if a work was run 3 hours ago and had 8 hours long period, after the update it would be still eligible for run in 5 hours, assuming that periodicity wasn't updated.

[`CANCEL_AND_REENQUEUE`](https://developer.android.com/reference/androidx/work/ExistingPeriodicWorkPolicy#CANCEL_AND_REENQUEUE) would do what you're describing (which I agree would be undesirable).

Reply:
Oh okay, I should better read the documentation next time ;)

Reply:
The worker isn't enqueued in `MainActivity`, so I assume you mean to refer to the application class.

I'm not sure where you're seeing an issue or how the flag solves it? It's fine to re-enqueue an existing periodic worker; that's what `ExistingPeriodicWorkPolicy` is for.

Just noting this for future reference. I'd like to get this merged soon, so I can change that myself later.